src/renderers/shared/reconciler/ReactCompositeComponent.js

```diff
@@ -195,6 +196,18 @@ var ReactCompositeComponentMixin = {
           Component.displayName || Component.name || 'Component'
         );
       }
+
+      var propsMutated = !shallowEqual(inst.props || {}, publicProps);
+      var componentName = Component.displayName || Component.name ||
+            'Component';
+
+      warning(
+        typeof inst.props === 'undefined' ||
```

Comment:
`inst.props === undefined` is simpler – but why not do this check in the React.Component constructor instead?